pkg/api/customization/app/app.go

```diff
@@ -38,6 +38,21 @@ func Formatter(apiContext *types.APIContext, resource *types.RawResource) {
 			delete(status, "lastAppliedTemplate")
 		}
 	}
+	var workloads []projectv3.Workload
+	if err := access.List(apiContext, &projectschema.Version, projectv3.WorkloadType, &types.QueryOptions{}, &workloads); err == nil {
+		for _, w := range workloads {
+			_, appID := ref.Parse(resource.ID)
+			if w.WorkloadLabels[appLabel] == appID {
+				resource.Values["state"] = w.State
```

Comment:
if there are more than one workload, the last value for state/transitioning will override the previous ones? Is it by intent? Looks like we should do some sort of aggregation. At least if one of the workload failed, the app should reflect it.